contractcourt/chain_arbitrator.go

```diff
@@ -1204,5 +1210,62 @@ func (c *ChainArbitrator) SubscribeChannelEvents(
 	return watcher.SubscribeChannelEvents(), nil
 }
 
+// FindOutgoingHTLCDeadline returns the deadline in absolute block height for
+// the specified outgoing HTLC. For an outgoing HTLC, its deadline is defined
+// by the timeout height of its corresponding incoming HTLC - this is the
+// expiry height the that remote peer can spend his/her outgoing HTLC via the
+// timeout path.
+func (c *ChainArbitrator) FindOutgoingHTLCDeadline(chanPoint wire.OutPoint,
```

Comment:
So the circuit map doesn't actually have the timeout information. Instead it only has the commitment height and HTLC index information: https://github.com/lightningnetwork/lnd/blob/b1175514f9854b965d4716ac015ea15387c63081/htlcswitch/circuit.go#L25-L64

So you _could_ look up by payment hash, then use _that_ to get the channel ID: https://github.com/lightningnetwork/lnd/blob/b1175514f9854b965d4716ac015ea15387c63081/channeldb/models/channel.go#L24-L40

Then use that to map the channel ID back into an outpoint to look up in the chain arb. 

We can add this information to the circuit map, but it'll need a migration, and also isn't actually needed for forwarding/resolution. 

Each time we revoke a state, before the circuit map is even updated, the active channel arbs will reflect the latest HTLC set: https://github.com/lightningnetwork/lnd/blob/b1175514f9854b965d4716ac015ea15387c63081/htlcswitch/link.go#L2204-L2216

Updating the state is a fully synchronous event.

Reply:
> o/w I'm pretty sure there's a griefing vector here where somebody can make you use a really urgent RefundTimeout by sending an HTLC to you with the same rhash but w/ a sooner deadline

Sending us that type of HTLC has _some_ cost as they need to have the funds to send it, and also be ready to go on chain in case something actually confirms. 

Initially I also advised YY to look into the circuit map, but then realized the information we want isn't actually there (see comment above), it it should be there in a consistent state in the chain arb. The MPP thing (can be many incoming HTLCs across many chans) is indeed a wart, but to address that we'd need to either:
  * Extend the circuit map with the extra information. 
  * Add another method to handle locating the precise HTLC that created the incoming circuit. 
  * Use our knowledge of the current routing policies to derive which incoming HLTC is the right one based on fees and input+output amount.

Reply:
Resolving as latest iteration uses the circuit map here.